File: 147_InsetionSortList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        temp = head
        toSort = []
        
        while temp:
            toSort.append(temp.val)
            temp = temp.next
        
        #print(toSort)
        
        #Use Insertion Sort to sort the number
        toSort.sort()
        
        dummy = ListNode()
        
        temp2 = dummy
        
        for i in range(len(toSort)):
            temp2.next = ListNode(toSort[i])
            temp2 = temp2.next
        
        
        return dummy.next

File: 147_InsetionSortList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* insertionSortList(ListNode* head) {
        
        vector<int> toSort;
        
        ListNode *temp = head;
        
        while(temp)
        {
            toSort.push_back(temp->val);
            temp = temp->next;
        }
        
        
        /*Use Insertion Sort to Sort the Numbers*/
        sort(toSort.begin(), toSort.end());
        
        ListNode *dummy = new ListNode();
        
        ListNode *temp2 = dummy;
        
        for(int i = 0; i < toSort.size(); i++)
        {
            temp2->next = new ListNode(toSort.at(i));
            temp2 = temp2->next;
        }
        
        return dummy->next;
            
        
    }
};